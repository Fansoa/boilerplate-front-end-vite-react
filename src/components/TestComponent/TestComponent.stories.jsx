import TestComponent from "@/src/components/TestComponent";

export default {
  title: "Component/TestComponent",
  component: TestComponent,
  args: {
    label: "label",
  },
};

export const TestComponentStory = {};
