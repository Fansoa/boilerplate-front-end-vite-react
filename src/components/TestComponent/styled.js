import styled from "@emotion/styled";

export const Container = styled.button`
  ${({ theme, isHover }) => `
    background-color: ${isHover ? theme.TestComponentTheme.hover.backgroundColor : theme.TestComponentTheme.default.backgroundColor};
  `}
`;

export const Label = styled.p`
  font-weight: 900;
`;
