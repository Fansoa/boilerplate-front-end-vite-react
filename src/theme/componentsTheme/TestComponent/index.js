import global from "../../global";

const TestComponentTheme = {
  default: {
    backgroundColor: global.colors.blue,
  },
  hover: {
    backgroundColor: global.colors.red,
  },
};

export default TestComponentTheme;
