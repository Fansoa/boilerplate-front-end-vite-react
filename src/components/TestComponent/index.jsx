import { useState } from "react";
import PropTypes from "prop-types";
import { Container, Label } from "./styled";

const TestComponent = ({ label }) => {
  const [isHover, setIsHover] = useState(false);

  return (
    <Container
      onMouseEnter={() => setIsHover(true)}
      onMouseLeave={() => setIsHover(false)}
      isHover={isHover}
    >
      <Label>{label}</Label>
    </Container>
  );
};

TestComponent.propTypes = {
  label: PropTypes.string.isRequired,
};

export default TestComponent;
