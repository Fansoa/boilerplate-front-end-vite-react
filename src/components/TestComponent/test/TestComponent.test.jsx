import { screen } from "@testing-library/react";
import TestComponent from "@/src/components/TestComponent";
import renderWithIntegrationProvider from "@/configs/vitest/customRender";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";

describe("TestComponent", async () => {
  it("Label is displayed", async () => {
    const { container } = renderWithIntegrationProvider(
      <TestComponent label="label" />,
    );
    const label = screen.getByText("label");
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("Button has been clicked", async () => {
    const { container } = renderWithIntegrationProvider(
      <TestComponent label="label" />,
    );
    const user = userEvent.setup();
    await act(async () => {
      await user.click(screen.getByRole("button", { name: /label/i }));
    });
    const toastMessage = screen.getByText(/Button has been clicked/i);
    expect(toastMessage).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
