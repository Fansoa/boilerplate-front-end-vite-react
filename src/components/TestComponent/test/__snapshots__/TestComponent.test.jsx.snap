// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label is displayed 1`] = `
.emotion-0 {
  background-color: blue;
}

.emotion-1 {
  font-weight: 900;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      label
    </p>
  </div>
</div>
`;
